node/src/components/storage.rs

```diff
@@ -1116,6 +1159,29 @@ impl Storage {
     fn mark_block_complete(&mut self, block_height: u64) -> Result<(), FatalStorageError> {
         self.completed_blocks.insert(block_height);
         self.persist_completed_blocks()?;
+        if self
+            .completed_blocks
+            .highest_sequence()
+            .map(|seq| {
+                seq.low() == 0
+                    // This next bit might not be necessary actually because
+                    // the highest sequence starting with 0 means we have
+                    // complete blocks starting from genesis with no gaps, and
+                    // the actual height of the tip might not give us any
+                    // relevant information, as it's constantly changing with
+                    // the chain progressing.
+                    && seq.high()
+                        == self
+                            .block_height_index
+                            .keys()
+                            .last()
+                            .copied()
+                            .unwrap_or_default()
+            })
+            .unwrap_or_default()
+        {
+            // Notify resync is complete.
+        }
```

Comment:
This does nothing, right? Shall we add a code to notify?

However, I'm not sure why we need some code. My assumption is that our current mechanisms should properly handle the state changes according to the content of a block store and select correct KeepUp/CatchUp/Validate state as the blocks are retrieved.

Reply:
It does nothing indeed. This is a remnant of a previous approach and I forgot to delete it. Right now, there is no point in notifying if a force resync has completed.